client/components/drop-zone/index.jsx

```diff
@@ -144,6 +153,25 @@ export const DropZone = React.createClass( {
 			// from tracked nodes since another "real" event will be triggered.
 			this.dragEnterNodes = without( this.dragEnterNodes, window );
 		}
+
+		this.toggleDropZoneReduxState( this.state.isDraggingOverDocument || this.state.isDraggingOverElement );
+	},
+
+	toggleDropZoneReduxState( isVisible ) {
+		/**
+		 * The DropZone component is included in components/libraries which are not yet reduxified,
+		 * for which `connect` ( from `react-redux` ) complains that it can't find `store`
+		 * in the current context.
+		 *
+		 * For places where the reduxification has happened, this will work.
+		 */
+		if ( this.context.store ) {
+			if ( isVisible ) {
+				this.context.store.dispatch( showDropZone() );
+			} else {
+				this.context.store.dispatch( hideDropZone() );
+			}
+		}
```

Comment:
There is a very (IMO) interesting conversation in #11739 about what I think you're trying to solve. You're effectively making the Redux connection optional, yes? By manually looking for `context.store` and only then attempting to dispatch.

In that other convo we agreed it would be an anti-pattern, so we tried different solutions.

Is your concern here that, by `connect`'ing this component, it would harm testability? Or something related? One common pattern is to have a component file export two things: the pure component and the connected one. I recommend you verify this, but IIRC the pure one is a named export and the connected one is the default export.

Let me know if your goal is something else!